### ISSUE TITLE
Compatibility with React-Native

### ISSUE THREAD

**Issue by ronhe** [GoogleChromeLabs/comlink#87]
Hi,
I have found Comlink very useful for RPC between a React-Native app and a web app inside a WebView component.
To make it work, I had to make a few adjustments:

1. Write an RN end-point that supports 3 functions: `AddEventListener`, `onMessage` that executes listeners and can be transferred to the WebView, and `send` that wraps the WebView's `postMessage` method.
2. Write a very simple web endpoint that just wraps `window.postMessage` and `document.addEventListener` into an object that supports `send` and `addEventListener`. That's because of the way RN injects the `postMessage` function, and since it only supports string messages.
3. Drop the call for `MessagePort` from the RN Comlink module, since it's undefined.
4. Transpile the Comlink ES6 module into ES5, since ES6 dependencies are unsupported by create-react-app.

I was wondering, if you think that Comlink should support this usage out of the box, or that it would be better to fork it and make the adaptations mentioned above.

Thanks!

**Comment by surma** [comment]
That sounds amazing!

It would be great if this worked out of the box. I’m not sure everything you did was actually necessary, but I’d be happy to help sort this out. For example, I already wrote `MessageChannelAdapter.ts` to support string-based channels (like WebRTC data channels), so that could probably be re-used there. I am very conscious of Comlink’s size, so obviously I aim to add as little special cases to the core library as possible.

I have very little experience with React and React native, so I’d be super helpful if you opened a PR and gave me instructions on how to run a demo project of yours using Comlink. Then we can work through your changes and figure out which ones are necessary.

**Comment by ronhe** [comment]
It took some more effort that expected, but I ended up wrapping `comlinkjs` with React-Native's `WebView` into a new package [rn-webview-rpc](https://github.com/ronhe/rn-webview-rpc) that exports both a React `WebViewRpc` component, and a matching web endpoint. I hope you that find that OK.

After sorting out many React-Native problems, I think that there is one issue that should be addressed in the `comlinkjs` package:
Currently Comlink is provided either in ES6 or in UMD (or globally).
At the moment, ES6 dependencies are not supported by [Create React App](https://github.com/facebook/create-react-app) ([read more about it here](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#npm-run-build-fails-to-minify)).
Also, UMD modules seems not to play nicely with WebPack: I've encountered a similar warning as in [this thread](https://stackoverflow.com/questions/38392697/webpack-umd-critical-dependency-cannot-be-statically-extracted).
Hence, to make it easier on fellow developers, I think that Comlink should be also provided in an ES5/commonJS format.
